projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php

```diff
@@ -1148,7 +1148,13 @@ public static function migrate_post_type_rules() {
 					continue;
 				}
 
-				$opts      = $wp_registered_widgets[ $widget ];
+				$opts = $wp_registered_widgets[ $widget ];
+
+				// Sometimes callbacks can be stored as Closure objects, we can't retrieve an option name for that.
+				if ( ! isset( $opts['callback'] ) || ! is_array( $opts['callback'] ) ) {
```

Comment:
This looks like it should work to resolve the error. Although we might want to add `|| ! isset( $opts['callback'][0]->option_name )` just in case the callback is an array but `[0]` isn't a `WP_Widget` object.
```suggestion
				if ( ! isset( $opts['callback'] ) || ! is_array( $opts['callback'] ) || ! isset( $opts['callback'][0]->option_name ) ) {
```

OTOH, if we want to risk more work, I wonder if something like this might be better?
```php
$id_base = wp_parse_widget_id( $widget )['id_base'];
$widget_object = $wp_widget_factory->get_widget_object( $id_base );
if ( ! $widget_object ) {
    continue;
}

$instances = get_option( $widget_object->option_name );
```
(that's cribbed from core's `src/wp-includes/rest-api/endpoints/class-wp-rest-widgets-controller.php`)

Reply:
I love it, thanks! I was looking for a more universal way to warrant an object (or lack of), and couldn't find it for some reason.